### PR TITLE
hashlink: add x86_64-darwin compatibility

### PR DESCRIPTION
**Pull request by l0go**
###### Description of changes
This allows Hashlink to be installed on x86_64-darwin platforms. Since the JIT functionality of the virtual machine does not support aarch64, I did not include that platform. However, Hashlink also can convert to C which does work on aarch64, so it might be worth testing that.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/interpreters/hashlink/default.nix [hunk 1]
```diff
@@ -12,12 +12,13 @@
...
 , SDL2
 , sqlite
+, getconf
 }:
 
 stdenv.mkDerivation rec {
   pname = "hashlink";
   version = "1.12";
-
+
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hashlink/default.nix:21]:
```suggestion

```

File: pkgs/development/interpreters/hashlink/default.nix [hunk 2]
```diff
@@ -43,11 +44,17 @@ stdenv.mkDerivation rec {
     sqlite
   ];
 
+  nativeBuildInputs = [ getconf ];
+
+  postFixup = lib.optionalString stdenv.isDarwin ''
+    install_name_tool -change libhl.dylib $out/lib/libhl.dylib $out/bin/hl
+  '';
+
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hashlink/default.nix:52]:
```suggestion

```

**Comment by logo4poop** [comment]
@SuperSandro2000 Made the changes, should I squash the commits?

File: pkgs/development/interpreters/hashlink/default.nix [hunk 3]
```diff
@@ -12,12 +12,12 @@
 , pcre
 , SDL2
 , sqlite
+, getconf
 }:
 
 stdenv.mkDerivation rec {
   pname = "hashlink";
   version = "1.12";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hashlink/default.nix:20]:
```suggestion
  version = "1.12";

```

File: pkgs/development/interpreters/hashlink/default.nix [hunk 4]
```diff
@@ -43,11 +43,16 @@ stdenv.mkDerivation rec {
     sqlite
   ];
 
+  nativeBuildInputs = [ getconf ];
+
+  postFixup = lib.optionalString stdenv.isDarwin ''
+    install_name_tool -change libhl.dylib $out/lib/libhl.dylib $out/bin/hl
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/hashlink/default.nix:50]:
```suggestion
  '';

```

**Pull request merged**